File: LeetCode/Medium/0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
    l := 0
    r := 0
    n := len(nums)
    temp := nums[0]
    ret := 100001
    
    for r < n {
        if temp >= target {
            ret = min(ret, r - l + 1)
            temp -= nums[l]
            l++
        } else {
            r++
            if r < n { temp += nums[r] }
        }
    }
    
    if ret == 100001 { return 0 }
    
    return ret
}

func min(a, b int) int {
    if a < b { return a }
    return b
}

File: LeetCode/Medium/0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l = 0, r = 0, n = nums.length;
        int temp = nums[0];
        int ret = Integer.MAX_VALUE;
        
        while (r < n){
            if (temp >= target){
                ret = Math.min(ret, r - l + 1);
                temp -= nums[l++];
            } else {
                ++r;
                if (r < n) temp += nums[r];
            }
        }
        
        if (ret == Integer.MAX_VALUE) return 0;
        
        return ret;
    }
}